spyder/plugins/editor/widgets/codeeditor/lsp_mixin.py

```diff
@@ -0,0 +1,1330 @@
+        self._server_requests_timer.setInterval(self.LSP_REQUESTS_SHORT_DELAY)
+        self._server_requests_timer.timeout.connect(
+            self._process_server_requests)
+
+        # Code Folding
+        self.code_folding = True
+        self.update_folding_thread = QThread(None)
+        self.update_folding_thread.finished.connect(self.finish_code_folding)
+
+        # Autoformat on save
+        self.format_on_save = False
+        self.format_eventloop = QEventLoop(None)
+        self.format_timer = QTimer(self)
+        self.__line_number_before_format = 0
+
+        # Outline explorer
+        self.oe_proxy = None
+
+        # Diagnostics
+        self.update_diagnostics_thread = QThread(None)
+        self.update_diagnostics_thread.run = self.set_errors
+        self.update_diagnostics_thread.finished.connect(
+            self.finish_code_analysis)
+        self._diagnostics = []
+
+        # Text diffs across versions
+        self.differ = diff_match_patch()
+        self.previous_text = ''
+        self.patch = []
+        self.leading_whitespaces = {}
+
+        # Other attributes
+        self.filename = None
+        self.completions_available = False
+        self.text_version = 0
+        self.save_include_text = True
+        self.open_close_notifications = True
+        self.sync_mode = TextDocumentSyncKind.FULL
+        self.will_save_notify = False
+        self.will_save_until_notify = False
+        self.enable_hover = True
+        self.auto_completion_characters = []
+        self.resolve_completions_enabled = False
+        self.signature_completion_characters = []
+        self.go_to_definition_enabled = False
+        self.find_references_enabled = False
+        self.highlight_enabled = False
+        self.formatting_enabled = False
+        self.range_formatting_enabled = False
+        self.document_symbols_enabled = False
+        self.formatting_characters = []
+        self.completion_args = None
+        self.folding_supported = False
+        self._folding_info = None
+        self.is_cloned = False
+        self.operation_in_progress = False
+        self.formatting_in_progress = False
+        self.symbols_in_sync = False
+        self.folding_in_sync = False
+
+    # ---- Helper private methods
+    # -------------------------------------------------------------------------
+    def _process_server_requests(self):
+        """Process server requests."""
+        # Check if the document needs to be updated
+        if self._document_server_needs_update:
+            self.document_did_change()
+            self.do_automatic_completions()
+            self._document_server_needs_update = False
+
+        # Send pending requests
+        for method, params, requires_response in self._pending_server_requests:
+            self.emit_request(method, params, requires_response)
+
+        # Clear pending requests
+        self._pending_server_requests = []
+
+    # ---- LSP: Basic methods
```

Comment:
```suggestion
    # ---- Basic methods
```

We don't need to mention LSP here.